tests/storages_tests/test_grpc.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+import pytest
+
+from optuna.storages import BaseStorage
+from optuna.study._study_direction import StudyDirection
+from optuna.testing.storages import STORAGE_MODES
+from optuna.testing.storages import StorageSupplier
+from optuna.trial import TrialState
+
+
+def _test_set_and_get_compatibility(
+    storage_set: BaseStorage, storage_get: BaseStorage, values: list[float] | None
+) -> None:
+    study_id = storage_set.create_new_study(directions=[StudyDirection.MINIMIZE])
+    trial_id = storage_set.create_new_trial(study_id)
+    assert storage_get.get_trial(trial_id).state == TrialState.RUNNING
+    storage_set.set_trial_state_values(trial_id, state=TrialState.COMPLETE, values=values)
+    assert storage_get.get_trial(trial_id).state == TrialState.COMPLETE
+    assert storage_get.get_trial(trial_id).values == values
+
+
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
```

Comment:
I have a question. `STORAGE_MODES` includes `grpc_rdb` and `grpc_journal_file`, so this test case currently runs the scenario that runs grpc storage proxy over grpc storage proxy?

Reply:
Yes. There is no particular need to exclude that cases, so I leave it as it is.

Reply:
I see. Thank you for your explanation.